evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +125,41 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectByType(ColumnSelector):
+    """
+    Selects columns by specified Woodwork logical type or semantic tag in input data.
+
+    Arguments:
+        column_types (string, ww.LogicalType, list(string), list(ww.LogicalType)): List of Woodwork types or tags, used to determine which columns to select.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Select Columns By Type Transformer"
+    hyperparameter_ranges = {}
+    """{}"""
+    needs_fitting = False
+    _VALIDATION_ERROR_FORMAT_STRING = (
+        "Columns of type {column_types} not found in input data."
+    )
+
+    def __init__(self, column_types=None, random_seed=0, **kwargs):
+        parameters = {"column_types": column_types}
+        parameters.update(kwargs)
+        Transformer.__init__(
```

Comment:
This, and the following comment, has me questioning if it's worth subclassing `ColumnSelector` rather than just `Transformer`; it seems like we're not gaining much value from `ColumnSelector` 😂

Reply:
That's true, the way this is written there's not a lot of benefit to having `SelectByType` subclass this implementation of `ColumnSelector`. That being said, it still makes logical sense to have it be a subclass, since it does function to select columns. With that, I see this more as a reason to consider refactoring `ColumnSelector` rather than removing `SelectByType` as a subclass.